sdk/options.go

```diff
@@ -0,0 +1,101 @@
+package tfsdklog
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+// Option defines a modification to the configuration for a logger.
+type Option func(loggerOpts) loggerOpts
+
+type loggerOpts struct {
+	name            string
+	level           hclog.Level
+	includeLocation bool
+
+	// some private options to be used only by tflog for testing purposes
+	// we should never export an Option that sets these
+	output      io.Writer
+	includeTime bool
+}
+
+func applyLoggerOpts(opts ...Option) loggerOpts {
+	// set some defaults
+	l := loggerOpts{
+		name:            "sdk",
```

Comment:
Given we only use this string once, and there's no inherent connection (as far as I know?) between the key to look up the logger in context and the name of the logger in the output, I'm not clear on what benefit this would yield.

Reply:
It's a generalization of avoiding https://en.wikipedia.org/wiki/Magic_number_(programming), but this is not a nit to hang my hat on.